qr_generator/lib/screens/qr_details/qr_details.dart

```diff
@@ -0,0 +1,138 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_countdown_timer/index.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qr_generator/screens/qr_details/bloc/qr_details_bloc.dart';
+import 'package:qr_generator/screens/widgets/error_view.dart';
+import 'package:qr_generator/screens/widgets/offline_view.dart';
+
+class QrDetails extends StatefulWidget {
+  QrDetails({Key? key}) : super(key: key);
+
+  @override
+  _QrDetailsState createState() => _QrDetailsState();
+}
+
+class _QrDetailsState extends State<QrDetails> {
+  late ThemeData _theme;
+  late QrDetailsBloc _qrDetailsBloc;
+  final Connectivity _connectivity = Connectivity();
+  late StreamSubscription<ConnectivityResult> _connectivitySubscription;
+
+  @override
+  void initState() {
+    super.initState();
+    _qrDetailsBloc = BlocProvider.of<QrDetailsBloc>(context);
+    _connectivitySubscription =
+        _connectivity.onConnectivityChanged.listen(_updateConnectionStatus);
+  }
+
+  @override
+  void dispose() {
+    _connectivitySubscription.cancel();
+    super.dispose();
+  }
+
+  Future<void> _updateConnectionStatus(ConnectivityResult result) async {
+    _qrDetailsBloc.add(CheckConnectivity(result));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _theme = Theme.of(context);
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Qr Details'),
+        actions: [
+          IconButton(
+              onPressed: () => _qrDetailsBloc.add(RefreshSeed()),
+              icon: const Icon(Icons.refresh))
+        ],
+      ),
+      body: SafeArea(
+        child: BlocListener(
```

Comment:
**fix**: This is not required! But if you do require a listener and a builder, use `BlocConsumer` instead of nesting widgets

Reply:
Done!